probe-rs/src/debug/variable_cache.rs

```diff
@@ -216,29 +275,33 @@ impl VariableCache {
     /// If there is more than one, it will be logged (tracing::warn!), and only the first will be returned.
     /// It is possible for a hierarchy of variables in a cache to have duplicate names under different parents.
     pub fn get_variable_by_name(&self, variable_name: &VariableName) -> Option<Variable> {
-        let child_variables = self
+        let mut child_variables = self
             .variable_hash_map
             .values()
             .filter(|child_variable| child_variable.name.eq(variable_name))
-            .cloned()
-            .collect::<Vec<Variable>>();
-        match child_variables.len() {
-            0 => None,
-            1 => child_variables.first().cloned(),
-            child_count => {
+            .collect::<Vec<&Variable>>();
+
+        // Sort the variables by key, so that we can consistently return the first one.
+        // The hash map does not return the values in a consistent order.
+        child_variables.sort_by_key(|v| v.variable_key);
```

Comment:
Why not use a BTreeMap?

Reply:
Good idea, didn't think of that.

(Was thinking of maybe replacing the `HashMap` completly, but using a `BTreeMap` makes a lot of sense with the current implementation.)